src/Auth0.php

```diff
@@ -298,6 +298,15 @@ public function exchange() {
       return false;
     }
 
+    $state = $this->getState();
+    if (!$state) {
+      return false;
+    }
+
+    if($this->store->get("state") != $state) {
```

Comment:
@glena What should we do if there is no stored value? As the developer does not have to use a store. I mean you have to go out of your way to not have a store as the default is `SessionStore`. Thoughts?

Reply:
what if we abstract that to a `StateHandler` (or whatever name) which will be responsible of generating and validatin the state.

if you want to handle on your own, you can use your implementation

```
inteface StateHandler {
public function __construct(SessionStore $store);
public function issue();
public function validate($state);
}
```

Reply:
by default, it will use our implementation that generates a random hash and stores that in session.
when it comes back validates it and if it does not match it will fail (throw new CoreException('invalid state')).